src/controllers/whatsapp-cache-management.ts

```diff
@@ -0,0 +1,110 @@
+import { logger } from '../utils/logger'
+import { options } from '../config'
+import { CreateConfig } from '../config/create-config'
+
+import * as fs from 'fs'
+import { HTTPRequest, Page } from 'puppeteer'
+
+class WhatsappCacheManagement {
+  private baseWppCacheURL =
+    'https://raw.githubusercontent.com/wppconnect-team/wa-version/main/html'
+  private waPage: Page | undefined
+  private waVersionHTML: string = ''
+  private webVersion: string = ''
+  private forceWebpack: boolean = false
+
+  async setup(waPage: Page, options: options | CreateConfig) {
```

Comment:
Não é necessário o async

```suggestion
  async setup(waPage: Page, options: options | CreateConfig) {
```